101-backup-create-storage-account-enable-protection/azuredeploy.json

```diff
@@ -0,0 +1,173 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vaultName": {
+      "type": "string",
+      "defaultValue": "[concat('vault', uniquestring(resourceGroup().id))]",
+      "metadata": {
+        "description": "Name of the Vault"
+      }
+    },
+    "vaultStorageRedundancy": {
+      "type": "string",
+      "defaultValue": "GeoRedundant",
+      "allowedValues": [
+        "LocallyRedundant",
+        "ZonallyRedundant",
+        "GeoRedundant"
+      ],
+      "metadata": {
+        "description": "Change Vault Storage Type (not allowed if the vault has registered backups)"
+      }
+    },
+    "backupPolicyName": {
+      "type": "string",
+      "defaultValue": "[concat('policy', uniquestring(resourceGroup().id))]",
+      "metadata": {
+        "description": "Name of the Backup Policy"
+      }
+    },
+    "retentionDays": {
+      "type": "int",
+      "defaultValue": 30,
+      "minValue": 1,
+      "maxValue": 360,
+      "metadata": {
+        "description": "Retention duration in days"
+      }
+    },
+    "storageAccountName": {
+      "type": "string",
+      "defaultValue": "[concat('store', uniquestring(resourceGroup().id))]",
+      "metadata": {
+        "description": "Name of the Storage Account"
+      }
+    },
+    "roleNameGuid": {
+      "type": "string",
+      "defaultValue": "[newGuid()]",
+      "metadata": {
+          "description": "A new GUID used to identify the role assignment"
+      }
+    },    
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources"
+      }
+    }
+  },
+  "variables": {
+    "roleDefinitionId": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/', 'e5e2a7ff-d759-4cd2-bb51-3152d37e2eb1')]",
```

Comment:
remove concat and use subscriptionResourceId

Reply:
Fixed